cognite/client/_api/datapoints.py

```diff
@@ -411,7 +417,128 @@ def retrieve_dataframe(
+            granularity (str): The granularity to fetch aggregates at. e.g. '1s', '2h', '10d'.
+            id (Union[int, List[int], Dict[str, Any], List[Dict[str, Any]]]: Id or list of ids. Can also be object
+                specifying aggregates. See example below.
+            external_id (Union[str, List[str], Dict[str, Any], List[Dict[str, Any]]]): External id or list of external
+                ids. Can also be object specifying aggregates. See example below.
+            limit (int): Maximum number of datapoints to return for each time series.
+            complete (str): post-processing of the dataframe.
+                Pass 'fill' to insert missing entries into the index, and complete data where possible (supports interpolation, stepInterpolation, count, sum, totalVariation).
+                Pass 'fill,dropna' to additionally drop rows in which any aggregate for any time series has missing values (typically rows at the start and end for interpolation aggregates).
+                This guarantees that all returned dataframes have the exact same shape and no missing values anywhere, and is only supported for aggregates using any of sum, count, totalVariance, interpolation and stepInterpolation
+
+        Returns:
+            Union[pandas.DataFrame,Dict[pandas.DataFrame]]: The requested dataframe if one aggregate was requested, or a dictionary of aggregate: dataframe if multiple aggregates were requested.
+
+        Examples:
+
+            Get a pandas dataframe without 'average' in the columns::
+
+                >>> from cognite.client import CogniteClient
+                >>> c = CogniteClient()
+                >>> df = c.datapoints.retrieve_dataframe_dict(id=[1,2,3], start="2w-ago", end="now",
+                ...         aggregates=["average"], granularity="1h")
+
+            Get a dictionary of pandas dataframes, with the index evenly spaced at 1h intervals, missing values completed in the middle and dropped at the ends::
+
+                >>> from cognite.client import CogniteClient
+                >>> c = CogniteClient()
+                >>> dfs = c.datapoints.retrieve_dataframe_dict(id=[1,2,3], start="2w-ago", end="now",
+                ...          aggregates=["interpolation","count"], granularity="1h", complete="fill,dropna")
+        """
+        all_aggregates = aggregates
+        for queries in [id, external_id]:
+            if isinstance(queries, list) and queries and isinstance(queries[0], dict):
+                for it in queries:
+                    for ag in it.get("aggregates") or []:
+                        if ag not in all_aggregates:
+                            all_aggregates.append(ag)
+
+        df = self.retrieve_dataframe(start, end, aggregates, granularity, id, external_id, limit, complete)
+        df_dict = {ag: df.filter(like="|" + ag).rename(columns=lambda s: s[: -len(ag) - 1]) for ag in all_aggregates}
+        return df_dict[aggregates[0]] if len(aggregates) == 1 else df_dict
```

Comment:
This should just always return the dict. The name of the method is retrieve_dataframe_dict, so it feels unnatural that it does this unpacking when there is only value in that dict.

Reply:
happy to rename the method while keeping this functionality - suggestions?